db/db_with_timestamp_basic_test.cc

```diff
@@ -296,34 +348,60 @@ TEST_F(DBBasicTestWithTimestamp, ForwardIterateStartSeqnum) {
   for (size_t i = 0; i != write_ts_list.size(); ++i) {
     Slice write_ts = write_ts_list[i];
     write_opts.timestamp = &write_ts;
-    uint64_t k = kMinKey;
-    do {
-      Status s = db_->Put(write_opts, Key1(k), "value" + std::to_string(i));
-      ASSERT_OK(s);
-      if (k == kMaxKey) {
-        break;
+    for (uint64_t k = kMaxKey; k >= kMinKey; --k) {
+      Status s;
+      if (k % 2) {
+        s = db_->Put(write_opts, Key1(k), "value" + std::to_string(i));
+      } else {
+        s = db_->Delete(write_opts, Key1(k));
       }
-      ++k;
-    } while (k != 0);
+      ASSERT_OK(s);
+    }
     start_seqs.push_back(db_->GetLatestSequenceNumber());
   }
   std::vector<std::string> read_ts_list;
   for (int t = 0; t != kNumTimestamps - 1; ++t) {
     read_ts_list.push_back(Timestamp(2 * t + 3, /*do not care*/ 17));
   }
+
   ReadOptions read_opts;
+  // Scan with only read_opts.iter_start_seqnum set.
   for (size_t i = 0; i != read_ts_list.size(); ++i) {
     Slice read_ts = read_ts_list[i];
     read_opts.timestamp = &read_ts;
-    read_opts.iter_start_seqnum = start_seqs[i];
+    read_opts.iter_start_seqnum = start_seqs[i] + 1;
     std::unique_ptr<Iterator> iter(db_->NewIterator(read_opts));
-    SequenceNumber expected_seq = start_seqs[i] + 1;
+    SequenceNumber expected_seq = start_seqs[i] + (kMaxKey - kMinKey) + 1;
     uint64_t key = kMinKey;
     for (iter->Seek(Key1(kMinKey)); iter->Valid(); iter->Next()) {
-      CheckIterEntry(iter.get(), Key1(key), expected_seq, kTypeValue,
+      CheckIterEntry(iter.get(), Key1(key), expected_seq,
+                     (key % 2) ? kTypeValue : kTypeDeletionWithTimestamp,
+                     "value" + std::to_string(i + 1), write_ts_list[i + 1]);
+      ++key;
+      --expected_seq;
+    }
+  }
+  // Scan with both read_opts.iter_start_seqnum and read_opts.iter_start_ts set.
+  std::vector<std::string> read_ts_lb_list;
+  for (int t = 0; t < kNumTimestamps - 1; ++t) {
+    read_ts_lb_list.push_back(Timestamp(2 * t, /*do not care*/ 17));
+  }
+  for (size_t i = 0; i < read_ts_list.size(); ++i) {
+    Slice read_ts = read_ts_list[i];
+    Slice read_ts_lb = read_ts_lb_list[i];
+    (void)read_ts_lb;
```

Comment:
Nit: probably left over/unnecessary line?

Reply:
Sorry about this. Will remove.